packages/reporter/src/MultiClient.ts

```diff
@@ -0,0 +1,39 @@
+import { createWalletClient, http, Chain, publicActions } from "viem"
+import { privateKeyToAccount } from "viem/accounts"
+
+type ContructorConfigs = {
+  chains: Chain[]
+  privateKey: `0x${string}`
+  rpcUrls: { [chainName: string]: string }
+}
+
+type GetClientsConfigs = {
+  chain?: Chain | undefined
+  privateKey: `0x${string}`
+  rpcUrl: string
+}
+
+const getClient = ({ chain, privateKey, rpcUrl }: GetClientsConfigs) =>
+  createWalletClient({
+    account: privateKeyToAccount(privateKey),
+    chain: chain as Chain | undefined,
+    transport: http(rpcUrl),
+  }).extend(publicActions)
+
+class Multiclient {
+  private _clients: { [chainName: string]: any }
```

Comment:
why `any`? shouldn't be `Client`?

Reply:
When declaring as `Client`, public & wallet actions like `simulateContract` and `writeContract`  cannot be accessed directly by the client instance in reporterController.ts after calling `getClient()`, even if we put the `.extend()`. Changing it to type `PublicClient & WalletClient` works.